src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/listener/OffenderEventListener.kt

```diff
@@ -27,6 +28,9 @@ class OffenderEventListener(
         upsertUnallocatedCaseService.upsertUnallocatedCase(crn)
       } catch (e: ForbiddenOffenderError) {
         log.warn("Unable to access offender with CRN $crn with error: ${e.message}")
+      } catch (e: EventsNotFoundError) {
+        log.warn("Unable to find events for CRN $crn with error: ${e.message}")
+        upsertUnallocatedCaseService.deleteEventsForNoActiveEvents(crn)
```

Comment:
The class suggests it's for saving/updating. The `deleteEventsForNoActiveEvents` method is for deleting.

Reply:
moved to the db operations class

Reply:
and added unit test